modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/CommandClosure.java

```diff
@@ -28,6 +28,14 @@
  * @see RaftGroupListener
  */
 public interface CommandClosure<R extends Command> {
+    /**
+     * Corresponding log index of the command. Present for write commands only.
+     * Returns {@code 0} for read commands.
+     */
+    default long appliedIndex() {
```

Comment:
Let's try to follow the original Raft paper and Jraft and name it lastAppliedIndex. That should be changed all over the code